operator/internal/controller/auto/workspace_controller.go

```diff
@@ -97,7 +99,9 @@ func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		w.Status.ObservedGeneration = w.Generation
 		ready.ObservedGeneration = w.Generation
 		meta.SetStatusCondition(&w.Status.Conditions, *ready)
-		return r.Status().Update(ctx, w)
+		// Use .Status().Patch() instead of .Update() to avoid conflicts with sequential reconcile loops hitting a
+		// stale version of the object from the cache.
+		return r.Status().Patch(ctx, w, client.MergeFrom(originalWorkspace), client.FieldOwner(FieldManager))
```

Comment:
We could alternatively do something like the following if we wanted to keep using `.Update` and reduce log spam:
```
err := r.Status().Update(...)
if err != nil && errors.IsConflict {
    return ctrl.Result{Requeue: true}, nil
}
return ctrl.Result{}, err
```

Reply:
Looking at [UpdateStatus](https://github.com/kubernetes/client-go/blob/def003bc9552f7e98726f5f2d05a701d33300418/gentype/type.go#L252-L265) makes me think patching (or more specifically put'ing) the status is fairly typical, so I'm OK with this.